optihood/energy_network.py

```diff
@@ -216,29 +213,37 @@ def get_nodal_data(func, data: _tp.Union[pd.ExcelFile, pd.DataFrame, dict]):
-            NodeKeys.solar.value: func(data, NodeKeys.solar.value),
-            NodeKeys.transformers.value: func(data, NodeKeys.transformers.value),
-            NodeKeys.demand.value: func(data, NodeKeys.demand.value),
-            NodeKeys.storages.value: func(data, NodeKeys.storages.value),
-            NodeKeys.stratified_storage.value: func(data, NodeKeys.stratified_storage.value),
-            NodeKeys.profiles.value: func(data, NodeKeys.profiles.value)
+            _ent.NodeKeys.buses.value: func(data, _ent.NodeKeys.buses.value),
+            _ent.NodeKeys.grid_connection.value: func(data, _ent.NodeKeys.grid_connection.value),
+            _ent.NodeKeys.commodity_sources.value: func(data, _ent.NodeKeys.commodity_sources.value),
+            _ent.NodeKeys.solar.value: func(data, _ent.NodeKeys.solar.value),
+            _ent.NodeKeys.transformers.value: func(data, _ent.NodeKeys.transformers.value),
+            _ent.NodeKeys.demand.value: func(data, _ent.NodeKeys.demand.value),
+            _ent.NodeKeys.storages.value: func(data, _ent.NodeKeys.storages.value),
+            _ent.NodeKeys.stratified_storage.value: func(data, _ent.NodeKeys.stratified_storage.value),
+            _ent.NodeKeys.profiles.value: func(data, _ent.NodeKeys.profiles.value)
         }
         return nodes_data
 
+    @staticmethod
+    def get_values_from_dataframe(df: pd.DataFrame, identifier: str, identifier_column: str, desired_column: str):
+        f""" Find any {identifier} entries in the {identifier_column} and return the 
+             {desired_column} value of those rows."""
+        row_indices = df[identifier_column] == identifier
+        values = df.loc[row_indices, desired_column].iloc[0]
+        return values
+
     def createNodesData(self, nodesData, file_or_folder_path, numBuildings, clusterSize):
         self.__noOfBuildings = numBuildings
 
         # update stratified_storage index
-        nodesData["stratified_storage"].set_index("label", inplace=True)
+        nodesData[_ent.NodeKeys.stratified_storage.value].set_index(_ent.StratifiedStorageLabels.label.value, inplace=True)
 
         # extract input data from CSVs
-        electricityImpact = nodesData["commodity_sources"].loc[nodesData["commodity_sources"]["label"] == "electricityResource", "CO2 impact"].iloc[0]
-        electricityCost = nodesData["commodity_sources"].loc[nodesData["commodity_sources"]["label"] == "electricityResource", "variable costs"].iloc[0]
-        demandProfilesPath = nodesData["profiles"].loc[nodesData["profiles"]["name"] == "demand_profiles", "path"].iloc[0]
-        weatherDataPath = nodesData["profiles"].loc[nodesData["profiles"]["name"] == "weather_data", "path"].iloc[0]
+        electricityImpact = self.get_values_from_dataframe(nodesData[_ent.NodeKeys.commodity_sources.value], _ent.CommoditySourceTypes.electricityResource.value, _ent.CommoditySourcesLabels.label.value, _ent.CommoditySourcesLabels.CO2_impact.value)
```

Comment:
First try of using the enums in the main code.
Currently this is very verbose.
Before using this throughout, we should discuss other options.